File: Visitor/PHP/ObjectStructure.php
<?php
  class ObjectStructure {
    protected $user_list;

    public function __construct() {
      $user_list = array();
    }

    public function addUser($user) {
      $this->user_list[] = $user;
    }

    public function handleVisitor($visitor) {
      foreach($this->user_list as $user) {
          $user->accept($visitor);
      }
    }
  }

File: Visitor/Go/visitor/ObjectStructure.go
package visitor

type ObjectStructure struct {
  user_list []UserApi
}

func (this *ObjectStructure) AddUser(user UserApi) {
  this.user_list = append(this.user_list, user)
}

func (this *ObjectStructure) HandleVisitor(visitor UserVisitor) {
  for _, user := range this.user_list {
    user.Accept(visitor)
  }
}
